src/testing/advection_explicit_periodic.cpp

```diff
@@ -10,180 +10,903 @@
+        q[dim-1] =p[dim-1] +  beta*std::sin(2.0 * pi * (p[dim-2])) * std::cos(pi /2.0 * p[dim-1]);
+
+        //Jesse Chan non sym transform
+
+//        q[0] =p[0] +beta* std::sin(pi*4.0 * (p[0])) * std::sin(pi * (p[1]+1));
+//        q[1] =p[1] -beta* 0.5*std::sin(pi*2.0 * (p[0])) * std::sin(pi /2.0 * (p[1]+1));
+
+        //Gassner nonsym
+
+//        q[0] =p[0] - 0.1*std::sin(2.0*pi*p[1]); 
+//        q[1] =p[1] + 0.1*std::sin(2.0*pi*p[0]);
+
+    }
+    if(dim==3){
+       //periodic non sym wild
+        double temp1 = alpha*(std::sin(pi * p[0]) * std::sin(pi * p[1]));
+        double temp2 = alpha*exp(1.0-p[1])*(std::sin(pi * p[0]) * std::sin(pi* p[1]));
+        q[0] =p[0] + temp1; 
+        q[1] =p[1] + temp2;
+        q[2] =p[2] +  1.0/20.0*( std::sin(2.0 * pi * q[0]) + std::sin(2.0 * pi * q[1]));
+
+    }
+
+    return q;
+}
+/****************************
+ * End of Curvilinear Grid
+ * ***************************/
+
+template<int dim, int nstate>
+double AdvectionPeriodic<dim, nstate>::compute_energy(std::shared_ptr < PHiLiP::DGBase<dim, double> > &dg) const
+{
+	double energy = 0.0;
+        dealii::LinearAlgebra::distributed::Vector<double> Mu_hat(dg->right_hand_side);
+        dg->global_mass_matrix.vmult( Mu_hat, dg->solution);
+        energy = dg->solution * Mu_hat;
+	return sqrt(energy);
+}
+
+template<int dim, int nstate>
+double AdvectionPeriodic<dim, nstate>::compute_conservation(std::shared_ptr < PHiLiP::DGBase<dim, double> > &dg, const double poly_degree) const
+{
+	double conservation = 0.0;
+       // dealii::LinearAlgebra::distributed::Vector<double> Mu_hat(dg->solution.size());
+        dealii::LinearAlgebra::distributed::Vector<double> Mu_hat(dg->right_hand_side);
+       // printf(" size Mass %d size sol %d\n",dg->global_mass_matrix.n(),dg->solution.size());
+       // printf("  Mass %g size sol %g\n",dg->global_mass_matrix(dg->global_mass_matrix.m(),dg->global_mass_matrix.n()),dg->solution(dg->solution.size()));
+        //dg->global_mass_matrix.vmult( Mu_hat, dg->right_hand_side);
+        dg->global_mass_matrix.vmult( Mu_hat, dg->solution);
+
+        const unsigned int n_dofs_cell = dg->fe_collection[poly_degree].dofs_per_cell;
+        const unsigned int n_quad_pts = dg->volume_quadrature_collection[poly_degree].size();
+        dealii::FullMatrix<double> Chi_operator(n_quad_pts, n_dofs_cell);
+        for(int istate=0; istate<nstate; istate++){
+            for (unsigned int itest=0; itest<n_dofs_cell; ++itest) {
+                for (unsigned int iquad=0; iquad<n_quad_pts; ++iquad) {
+                    const dealii::Point<dim> qpoint  = dg->volume_quadrature_collection[poly_degree].point(iquad);
+                    Chi_operator[iquad][itest] = dg->fe_collection[poly_degree].shape_value_component(itest,qpoint,istate);
+                }
+            }
+        }
+        dealii::FullMatrix<double> Chi_inv(n_dofs_cell);
```

Comment:
Avoid symbols. Everyone uses different symbols. Instead, call them what they are. Interpolation operators, vandermonde matrix, projection operator, mass matrix, stiffness matrix, filter matrix, etc.

I'm going to assume that the PR is peppered with symbols, not going to comment on all of them, please change those names.

Reply:
fixed